packages/vaex-core/vaex/arrow/convert.py

```diff
@@ -243,3 +243,32 @@ def trim_buffers_ipc(ar):
 def trim_buffers_for_pickle(ar):
     # future version of pyarrow might fix this, so we have a single entry point for this
     return trim_buffers_ipc(ar)
+
+
+def align(a, b):
```

Comment:
You may be interested in https://github.com/xhochy/fletcher/blob/4bf599207c018fafea40688c2d771320872c7c23/fletcher/algorithms/utils/chunking.py#L132-L142 and the functions it is calling. This is how I apply a binary (array-based) function to two chunked arrays in fletcher.

Reply:
Nice Uwe! Thanks for sharing this.

Reply:
I wonder if this should be upstreamed to arrow in the future.

Reply:
Definitely, there are some nice things in the `fletcher/(_)algoritms` module that make the general usage of `pyarrow`-in-Python users much simpler. These things are just aren't needed inside of Arrow yet and I'm still find the one or other bug / generalisation which keeps from contributing them *now*.

Reply:
keep me in the loop, would be nice to join forces, and benchmark all these things.